packages/taco/src/conditions/context/context.ts

```diff
@@ -133,43 +116,67 @@ export class ConditionContext {
     // TODO: Figure out if we can simplify/deduplicate this logic for every
     // authentication method we use
 
-    // Handle legacy EIP712 signature denoted by ":userAddress"
+    // Handle a legacy EIP712 signature denoted by ":userAddress"
     if (requestedParameters.has(USER_ADDRESS_PARAM_DEFAULT)) {
       if (!this.eip712SignatureProvider) {
         throw new Error(ERR_SIGNER_REQUIRED(USER_ADDRESS_PARAM_EIP712));
       }
       parameters[USER_ADDRESS_PARAM_DEFAULT] =
         await this.eip712SignatureProvider.getOrCreateWalletSignature();
-      // Remove from requested parameters
-      requestedParameters.delete(USER_ADDRESS_PARAM_DEFAULT);
     }
 
-    // Handle a new alias for EIP712, ":userAddressEIP712"
+    // Handle an alias for EIP712, ":userAddressEIP712"
     if (requestedParameters.has(USER_ADDRESS_PARAM_EIP712)) {
       if (!this.eip712SignatureProvider) {
         throw new Error(ERR_SIGNER_REQUIRED(USER_ADDRESS_PARAM_EIP712));
       }
       parameters[USER_ADDRESS_PARAM_EIP712] =
         await this.eip712SignatureProvider.getOrCreateWalletSignature();
-      // Remove from requested parameters
-      requestedParameters.delete(USER_ADDRESS_PARAM_EIP712);
     }
 
-    // Handle a new method, EIP4361, with a ":userAddressEIP4361" alias
+    // Handle a EIP4361 method, with a ":userAddressEIP4361" alias
     if (requestedParameters.has(USER_ADDRESS_PARAM_EIP4361)) {
       if (!this.eip4361SignatureProvider) {
         throw new Error(ERR_SIGNER_REQUIRED(USER_ADDRESS_PARAM_EIP4361));
       }
       parameters[USER_ADDRESS_PARAM_EIP4361] =
         await this.eip4361SignatureProvider.getOrCreateWalletSignature();
-      // Remove from requested parameters
-      requestedParameters.delete(USER_ADDRESS_PARAM_EIP4361);
+    }
+
+    // Handle a special case for EIP4361, with a ":userAddressExternalEIP4361" alias
+    // If we have both EIP4361 and an external EIP4361, we should throw
+    if (
+      requestedParameters.has(USER_ADDRESS_PARAM_EIP4361) &&
+      requestedParameters.has(USER_ADDRESS_PARAM_EXTERNAL_EIP4361)
+    ) {
+      throw new Error(
+        ERR_CONFLICTING_CONTEXT_PARAMS([
+          USER_ADDRESS_PARAM_EIP4361,
+          USER_ADDRESS_PARAM_EXTERNAL_EIP4361,
+        ]),
+      );
+    }
+
+    // Now, we need to handle a special case for external EIP4361
+    // We expect to find it in the custom parameters
+    if (requestedParameters.has(USER_ADDRESS_PARAM_EXTERNAL_EIP4361)) {
+      if (!this.customParameters[USER_ADDRESS_PARAM_EXTERNAL_EIP4361]) {
+        throw new Error(
+          ERR_MISSING_CONTEXT_PARAMS([USER_ADDRESS_PARAM_EXTERNAL_EIP4361]),
+        );
+      }
+      // Notice that we are using the same value for both EIP4361 and external EIP4361
+      parameters[USER_ADDRESS_PARAM_EIP4361] =
+        this.customParameters[USER_ADDRESS_PARAM_EXTERNAL_EIP4361];
```

Comment:
What's the reason for this? On L148-L151 we check to ensure that both entries are not in concurrently in the requested parameters.

Reply:
See https://github.com/nucypher/nucypher/pull/3513.

Reply:
Sure, we can address this on the server-side. This was a workaround for (at the time) not having this support.